tests/functional/report/checkRoutes.js

```diff
@@ -0,0 +1,94 @@
+'use strict'; // eslint-disable-line strict
+const http = require('http');
+const https = require('https');
+const fs = require('fs');
+
+const conf = require('../config.json');
+
+const transportStr = conf.transport;
+const transport = transportStr === 'http' ? http : https;
+
+function makeAgent() {
+    if (transportStr === 'https') {
```

Comment:
This currently would never be the case (only in the end to end). Do you think we should be testing this in just s3?